pkg/monitors/elasticsearch/query/request.go

```diff
@@ -0,0 +1,117 @@
+package query
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+// Represents HTTP post data for elasticsearch search queries
+// This format is generally determined by the format specified here:
+// https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations.html#_structuring_aggregations
+// Note that it also depends on the aggregations supported by the
+// query monitor
+type HTTPPostData struct {
+	// Holds a map of aggregation names to the corresponding aggregation
+	Aggregations map[string]aggregationInfo `json:"aggs"`
+}
+
+// Represents both, bucket and metric aggregations
+// See https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations.html
+// for details about various type of bucket and metric aggregations
+type aggregationInfo struct {
+	// Holds aggregationInfo type and properties specific to the aggregation
+	AggregationProps map[string]interface{} `json:"-"`
+	// Sub aggregations within the parent aggregation
+	SubAggregations map[string]aggregationInfo `json:"aggs,omitempty"`
+}
+
+type AggregationMeta struct {
```

Comment:
Is it preferable to use a struct with one string field rather than just using a string